core/dbt/events/event_handler.py

```diff
@@ -0,0 +1,40 @@
+import logging
+from typing import Union
+
+from dbt.events.base_types import EventLevel
+from dbt.events.types import Note
+
+from dbt.events.eventmgr import IEventManager
+
+_log_level_to_event_level_map = {
+    logging.DEBUG: EventLevel.DEBUG,
+    logging.INFO: EventLevel.INFO,
+    logging.WARN: EventLevel.WARN,
+    logging.WARNING: EventLevel.WARN,
+    logging.ERROR: EventLevel.ERROR,
+    logging.CRITICAL: EventLevel.ERROR,
+}
+
+
+class DbtEventLoggingHandler(logging.Handler):
+    """A logging handler that wraps the EventManager
+    This allows non-dbt packages to log to the dbt event stream.
+    All logs are generated as "Note" events.
+    """
+
+    def __init__(self, event_manager: IEventManager, level):
+        super().__init__(level)
+        self.event_manager = event_manager
+
+    def emit(self, record: logging.LogRecord):
+        note = Note(msg=record.getMessage())
+        level = _log_level_to_event_level_map[record.levelno]
```

Comment:
The previous approach had a default scenario. Do we not need that? i.e. every value of `record.levelno` is in `_log_level_to_event_level_map` and `record.levelno` is always populated?

Reply:
yes it should be populated by default